interfaces/Readme.md

```diff
@@ -0,0 +1,482 @@
+
+```java
+for (int i = 0; i < length; i++) {
+    list.add(i); // add will insert i at the end of the list.
+}
+```
+
+However, we need to time this.
+The easiest method for this is to use `System.currentTimeMillis()` which simply returns the current time as a `long` (a very big integer).
+The documentation for this function can be found [here](https://docs.oracle.com/javase/8/docs/api/java/lang/System.html#currentTimeMillis--).
+So, to time our function, all we must do is grab the current time before and after running our code.
+We can then take the difference of these times to find the actual elapsed time.
+
+```java
+long start_time = System.currentTimeMillis();
+
+for (int i = 0; i < length; i++) {
+    list.add(i);
+}
+
+long end_time = System.currentTimeMillis();
+System.out.printf("\tFilling List: %dms\n", end_time - start_time);
+```
+
+Now, we can go through our list again and try to increment every element.
+This is done in much the same way as before, except we need to both `get` an element from our list and then `set` it back after incrementing.
+
+```java
+for (int i = 0; i < length; i++) {
+
+    // Retrieve the value.
+    int value = list.get(i);
+
+    // Now set it back into the list.
+    list.set(i, value + 1);
+
+}
+```
+
+Just as above, we will time this part of the function by calling `currentTimeMillis` before and after our code runs.
+
+Lastly, we want to see how long it takes to search these lists.
+We will only look for a few items within the list to get a sense of the performance.
+For this, we can use the `contains` method of the `List` interface which will return a `boolean` based on whether the list contains our item.
+Let's look for a few elements near the start and the end of the list.
+
+```java
+boolean contains_1 = list.contains(1);
+boolean contains_5 = list.contains(5);
+boolean contains_length = list.contains(length);
+boolean contains_length_add_1 = list.contains(length + 1);
+```
+
+Once again, we will time this code, but we will also print out the results of the search.
+This big print statement will do just that.
+It prints each of our above booleans and the element they related to.
+
+```java
+System.out.printf(
+    "\t\tFound: 1=%b, 5=%b, %d=%b, %d=%b\n",
+    contains_1,
+    contains_5,
+    length,
+    contains_length,
+    length + 1,
+    contains_length_add_1
+);
+```
+
+Now, looking at our `time_list` method in full we have the following (provided in the [code folder](./code/Profiler.java)):
+
+```java
+public static void time_list(List<Integer> list, String name, int length) {
+
+    System.out.printf("Timing %s:\n", name);
+
+    // First, we fill the list with integers.
+    long start_time = System.currentTimeMillis();
+
+    for (int i = 0; i < length; i++) {
+        list.add(i);
+    }
+
+    long end_time = System.currentTimeMillis();
+    System.out.printf("\tFilling List: %dms\n", end_time - start_time);
```

Comment:
And the next couple as well, of course.

Reply:
Yes, I was thinking that towards the end of writing this. Was wary since it'd likely be static.
I could make it so the profiler is a class which takes the length as constructor argument and then the time function takes just the list / name.

Reply:
Yep, made `timeList` non static.